helix-core/src/main/java/org/apache/helix/model/InstanceConfig.java

```diff
@@ -501,14 +546,7 @@ private List<InstanceOperation> getInstanceOperations() {
     return _deserializedInstanceOperations;
   }
 
-  /**
-   * Set the instance operation for this instance.
-   * This method also sets the HELIX_ENABLED, HELIX_DISABLED_REASON, and HELIX_DISABLED_TYPE fields
-   * for backwards compatibility.
-   *
-   * @param operation the instance operation
-   */
-  public void setInstanceOperation(InstanceOperation operation) {
+  private void setInstanceOperationHelper(InstanceOperation operation) {
```

Comment:
Some question about the helper,
1. If the current operation is ENABLE, dont we substitute the DISABLE operation in the list to ENABLE instead? 
2. Should we write back ENABLE if HELIX_ENABLED is true and operation list is not enable? 
3. I think we can merge the 2 `operation.getOperation() == InstanceConstants.InstanceOperation.ENABLE`.